content/docs/going-deeper/tasks.md

```diff
@@ -151,7 +151,14 @@ call park, if an unpark happens it'll store `true` to `ready` (as we saw above
 in `Unpark for ThreadUnpark`). If this hasn't happened, though, then we block
 the current thread. The current thread will then get reawoken once the future's
 task is unparked, through the `ThreadUnpark` instance associated with this
-future.
+future. This must be done from another thread that is not parked. Some futures
+need a `Core` to be running in some thread to proceed, or some other trigger
+they registered into. Obviously a different one than which we block with `wait`.
```

Comment:
This comment is only about the `wait` executor, which calls `thread::park` internally. Unless I'm very mistaken, if you wait on a future, you can't run a `Core` at the same thread at that time.

Or is there some more magic I don't see?